src/auro-avatar.js

```diff
@@ -62,6 +64,10 @@ class AuroAvatar extends LitElement {
       },
       type: {
         type: String
+      },
+      httpStatus: {
+        type: String,
+        reflect: true
```

Comment:
I always forget... when the intent is only to expose the value to an outside consumer, e.g. `el.httpStatus`, I don't think we need to make this a prop? A prop is only needed when we want to pass info INTO the scope of the element. 

@geoffrich am I wrong on this?

Reply:
I see what you are doing. I am not sure putting the return into a prop on in the DOM is what a dev would want. If I remove the prop from the definition of the element, I can still do the following. 

<img width="457" alt="Screen Shot 2022-01-20 at 1 37 13 PM" src="https://user-images.githubusercontent.com/181089/150426471-983faa81-8827-4d8c-87aa-c4a5d5ef12d0.png">

<img width="403" alt="Screen Shot 2022-01-20 at 1 39 05 PM" src="https://user-images.githubusercontent.com/181089/150426524-0cd89dc8-bb64-4dbf-8649-cbee1ebe86d5.png">

Reply:
If we want to expose the returned status code to the consumer of this component, I think a custom event would be more appropriate. Otherwise, it's not clear _when_ to access the `httpStatus` prop, since the consumer doesn't know when the fetch completed.

Props are for passing data into the component, events are for passing data out of the component.

Reply:
I think we need to get clarity on the ask here. 

We discussed the original ask (regional fallback images) from @gusnaughton and decided not to do that inside avatar. Instead, we would provide a way for Gus to detect the failure himself and redo avatar with whatever fallback he wants.

Event was actually my original thought but now I think I want to just get agreement on the direction before making further changes. @braven112 did you come to any resolution on this with @gusnaughton?

Reply:
Per Auro team discussion I am reverting the property for httpStatus and we will resolve the notification of failures in a separate ticket.

Reply:
While I agree that using a prop to communicate out is not appropriate, I also don't think that we need to set a custom event to expose a data value? By setting `this.httpStatus = statusCode;` like in the example above, any dev can swing by and get the status value. This is no different than what we are doing with auro-input? 

Is there an advantage of setting a custom event over just setting the value that I am not seeing?

Reply:
Any dev can swing by, but the issue is knowing when to swing by. How do you know when the request failed and that you should provide a fallback? A custom event tells the consumer that something happened and they need to respond to it, otherwise you have to keep checking.

e.g. in Svelte, I render auro-avatar. I can bind to it to check httpStatus, but it won't be set right away -- there's latency involved in making the HTTP request. I need to know when the component is falling back to a default so I can take appropriate action. 

Here's what my component would look like if I had an "error" custom event that tells me the image failed:
```html
<script>
	let code = 'INVALID';
	
	function handleInvalidCode() {
		code = 'FALLBACK';
	}	
</script>

<auro-avatar code={code} on:error={handleInvalidCode}></auro-avatar>
```

And here's what I would have to do if I don't know when `httpStatus` is set. I would have to set a timeout to wait for the request to finish, and then check the property. This would be flaky because some requests could take longer than others. With the event, I am notified right away.

```html
<script>
	let code = 'INVALID';
	let avatar;
	
	function handleInvalidCode() {
		code = 'FALLBACK';
	}	
	
	setTimeout(() => {
		if (avatar.httpStatus === 404) {
			handleInvalidCode();
		}
	}, 500) // have to manually hardcode a delay here, which could be flaky	
</script>

<auro-avatar bind:this={avatar} code={code}></auro-avatar>
```

Reply:
This is the same reason why an `<input>` exposes an input event. Sure, the value is available as a property, but you want to know when that property was updated.

Reply:
@geoffrich thanks, that is what I was looking for!